stubs/mock/mock/mock.pyi

```diff
@@ -378,3 +379,13 @@ class PropertyMock(Mock):
     def __set__(self, obj: Any, value: Any) -> None: ...
 
 def seal(mock: Any) -> None: ...
+
+class ThreadingMixin(Base):
+    DEFAULT_TIMEOUT: float | None
+
+    def __init__(self, *args: Any, timeout: float | None = ..., **kwargs: Any) -> None: ...
+    def reset_mock(self, *args: Any, **kwargs: Any) -> None: ...
```

Comment:
This calls `super().reset_mock()`: https://github.com/testing-cabal/mock/blob/d344fa2794b3b1ae7e4a4dbf265fb040d6f41d1f/mock/mock.py#L3081

When this class is used as a superclass for `ThreadingMock`, that `super()` call ends up calling `NonCallableMock.reset_mock()`. So perhaps the signature of this method should match that method?

```suggestion
    def reset_mock(self, visited: Any = None, *, return_value: bool = False, side_effect: bool = False) -> None: ...
```

Reply:
In theory, it can be used with something else, since it is a mixin.
For example, `AsyncMockMixin` in the same file uses just `*args, **kwargs`

Reply:
I don't think this class is intended to be used as a mixin in conjunction with any classes outside `unittest.mock` or the third-party `mock` package, so I think this is just going to lead to unnecessarily imprecise typechecking. But you are strictly speaking correct.

As a compromise, what if we overrode `reset_mock()` on the `ThreadingMock` class (even though that's not what happens at runtime), with the more precise signature that you see on `NonCallableMock.reset_mock()`?

Reply:
> if we overrode reset_mock() on the ThreadingMock class (even though that's not what happens at runtime), with the more precise signature that you see on NonCallableMock.reset_mock()?

Yes, that's a good idea. I can do the same for `AsyncMockMixin` in the next PR.